util/info.c

```diff
@@ -171,6 +174,30 @@ uint64_t tokparse(char *caps, uint64_t (*str2flag) (char *inputstr))
 	return flags;
 }
 
+int print_vars() {
+	int ret, count;
+	struct fi_setting *env;
+
+	ret = fi_getsettings(&env, &count);
+
+	if (ret)
+		return ret;
+
+	for (int i = 0; i < count; ++i) {
+		printf("# %s: %s\n", env[i].prov_name, env[i].help_string);
+
+		if (env[i].value)
+			printf("%s=%s\n", env[i].env_var_name, env[i].value);
```

Comment:
Do you need to quote the second `%s` if there are spaces in `env[i].value`?

If the output of this script is intended to be `source`d, it would probably be a good idea.